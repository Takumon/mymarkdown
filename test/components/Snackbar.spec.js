import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex, { mapState, mapActions } from 'vuex'
import {
  MdSnackbar,
  MdButton,
} from 'vue-material/dist/components'

import TestTargetVue from '../../components/Snackbar.vue'


jest.useFakeTimers();

describe('Snackbar.vue', () => {

  let wrapper
  let store
  beforeEach(() => {
    const localVue = createLocalVue()
    localVue.use(MdSnackbar)
    localVue.use(MdButton)
    localVue.use(Vuex)

    const sysdate = new Date().toString()

    store = new Vuex.Store({
      state: {
        snackbarText: 'SampleText',
        showSnackbar: true,
      },
      actions: {
        setShowSnackbar ({ commit }, { isShow, text}) {
          return new Promise((resolve, reject) => {
            commit( 'setShowSnackbar', { isShow, text })
            resolve()
          })
        },
      },
      mutations: {
        setShowSnackbar (state, payload) {
          state.showSnackbar = payload.isShow
          // textを指定してる時のみ更新
          if (payload.text) {
            state.snackbarText = payload.text
          }
        },
      }
    })

    wrapper = shallowMount(TestTargetVue, {
      store,
      localVue,
    })
    jest.runAllTimers()
  })

  describe('初期表示', () => {

    test('スナップショット', () => {
      expect(wrapper).toMatchSnapshot()
    })

    test('スナックバーが表示されている', () => {
      const $text = wrapper.find('[data-test="snackbar-text"]')
      expect($text.text()).toBe('SampleText')
    })
  })

  describe('閉じるボタンクリック時', () => {
    beforeEach(() => {
      expect(store.state.showSnackbar).toBe(true)
      wrapper.find('[data-test="close-button"]').trigger('click')
      jest.runAllTimers()
    })

    test('スナックバーが表示されてない', () => {
      expect(store.state.showSnackbar).toBe(false)
    })
  })

})
