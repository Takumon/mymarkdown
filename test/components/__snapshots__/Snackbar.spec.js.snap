// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar.vue 初期表示 スナップショット 1`] = `
<div class="md-snackbar md-position-center md-theme-default" name="md-snackbar" appear="">
  <div class="md-snackbar-content"><span data-test="snackbar-text">SampleText</span>
    <button type="button" class="md-button md-theme-default" data-test="close-button">
      <div class="md-ripple">
        <div class="md-button-content">
          <i class="fas fa-times"></i>
        </div>
      </div>
    </button>
  </div>
</div>
`;
